doc/sf/jacobi_theta.qbk

```diff
@@ -0,0 +1,349 @@
+
+      template <class T>
+      ``__sf_result`` jacobi_theta1tau(T x, T tau);
+
+      template <class T, class Policy>
+      ``__sf_result`` jacobi_theta1tau(T x, T tau, const Policy&);
+  }} // namespaces
+
+[heading Description]
+
+The functions calculate the value of first [link math_toolkit.jacobi_theta.jacobi_theta_overview Jacobi Theta function], parameterized either in terms of the nome /q/:
+
+[equation jacobi_theta1] [/ \theta_1(x, q) = 2 \sum_{n=0}^\infty (-1)^n q^{(n+\frac{1}{2})^2} \sin((2n+1)x) ]
+
+Or in terms of an imaginary [tau]:
+
+[equation jacobi_theta1tau] [/ \theta_1(x|\tau) = 2 \sum_{n=0}^\infty (-1)^n \exp(i\pi\tau{(n+0.5)^2}) \sin((2n+1)x) ]
+
+The nome /q/ is restricted to the domain (0, 1), returning the result of __domain_error otherwise. The following graph shows the theta function at various values of /q/:
+
+[graph jacobi_theta1]
+
+[optional_policy]
+
+[heading Accuracy]
+
+The following [link math_toolkit.ulps_plots ULPs plot] is representative, fixing /q/=0.5 and varying /x/ from 0 to 2[pi]:
+
+[graph jacobi_theta1_float]
+
+The envelope represents the function's [link math_toolkit.cond condition number]. Note that relative accuracy degenerates periodically near [theta][sub 1]=0.
+
+Fixing /x/=5 and varying /q/, the ULPs plot looks like:
+
+[graph jacobi_theta1q_float]
+
+Accuracy tends to degenerate near /q/=1 (small [tau]).
+
+[heading Implementation]
+
+The /q/ parameterization is implemented using the [tau] parameterization, where [tau]=-log(/q/)/[pi].
+
+If [tau] is greater than or equal to 1, the summation above is used as-is. However if [tau] < 1, the following identity [@https://dlmf.nist.gov/20.7#viii DLMF 20.7.30] is used, defining [tau]'=-1/[tau]:
+
+[equation jacobi_theta1_imaginary] [/ (-i\tau)^{1/2}\theta_1(x|\tau)=-i\exp(i\tau'x^2/\pi)\theta_1(x\tau'|\tau') ]
+
+This transformation of variables ensures that the function will always converge in a small number of iterations.
+
+[endsect] [/section:jacobi_theta1 Jacobi Theta Function [theta][sub 1]]
+
+[section:jacobi_theta2 Jacobi Theta Function [theta][sub 2]]
+
+[heading Synopsis]
+
+``
+  #include <boost/math/special_functions/jacobi_theta.hpp>
+``
+
+  namespace boost { namespace math {
+      template <class T>
+      ``__sf_result`` jacobi_theta2(T x, T q);
+
+      template <class T, class Policy>
+      ``__sf_result`` jacobi_theta2(T x, T q, const Policy&);
+
+      template <class T>
+      ``__sf_result`` jacobi_theta2tau(T x, T tau);
+
+      template <class T, class Policy>
+      ``__sf_result`` jacobi_theta2tau(T x, T tau, const Policy&);
+  }} // namespaces
+
+[heading Description]
+
+The functions calculate the value of second [link math_toolkit.jacobi_theta.jacobi_theta_overview Jacobi Theta function], parameterized either in terms of the nome /q/:
+
+[equation jacobi_theta2] [/ \theta_2(x, q) = 2 \sum_{n=0}^\infty q^{(n+\frac{1}{2})^2} \cos((2n+1)x) ]
+
+Or in terms of an imaginary [tau]:
```

Comment:
I'm confused. You say you enforce that tau is purely imaginary, yet your function signature says `tau` is the same type as `x`. . .

Reply:
`tau` the C++ argument is real-typed. The provided number is implicitly multiplied by *i*. This seemed simpler than trying to provide a complex-valued API.